doc/in-progress/custom-upgrade-policy.md

```diff
@@ -0,0 +1,1000 @@
+# Custom Upgrade Policies
+
+`sui client upgrade` offers a simple way to upgrade packages when the
+CLI's active address owns the package's `UpgradeCap`.  This process is
+useful to get started with upgrades, or in the early stages of a
+package's development, but having the ability to upgrade a package be
+protected by only one key can pose a security risk for packages that
+are live in production:
+
+- The individual owning that key may make changes that are in their
+  interests but not the interests of the broader community.
+- Upgrades may happen without enough time for package users to consult
+  on the change or stop using the package if they disagree.
+- The key may get lost.
+
+Ideally when a package goes live it is made **immutable**.  This is
+done by using `sui::package::make_immutable` to burn its `UpgradeCap`,
+and prevents any further changes, but the reality is that in many
+cases package creators may want to retain an ability to upgrade their
+package in future to fix bugs, or add new features.  In those cases,
+some safety and security can be maintained for the package creator and
+users through **custom upgrade policies**.
+
+## Overview
```

Comment:
nit, and just a high level comment about this section, putting on my "as a user" hat, it might be worth emphasizing that the ticket/receipt system ensures only one upgrade happens in a transaction. To answer a basic question of "why do I need this tickets/receipts machinery and why should I worry about it? Can't I just use the UpgradeCap to do the upgrade?"

Reply:
Let me see if I can reword the introductory section to make this bit clearer -- I think if I try and motivate the tickets and receipts design directly, we'll get into the weeds quite quickly and it's too early on in the docs for that, but you're right that the motivation for why it's not sufficient to give whoever wants to upgrade the `UpgradeCap` is not clear here.

Incidentally, the purpose of tickets and receipts isn't to ensure that only one upgrade happens concurrently in a transaction, `UpgradeCap` already provides that property by default, and we had to do work to get it back when we introduced tickets and receipts.  Their purpose is to allow folks to provide limited access to an `UpgradeCap`'s ability to authorize upgrades without having to provide access to all its other privileges (changing compatibility bound, making the package immutable, etc).